src/main/java/com/example/bookstoreapp/controller/BookController.java

```diff
@@ -33,8 +37,21 @@ public ResponseEntity<BookDto> getBookById(@PathVariable Long id) {
     }
 
     @PostMapping
-    public ResponseEntity<BookDto> createBook(@RequestBody CreateBookRequestDto bookDto) {
+    public ResponseEntity<BookDto> createBook(@RequestBody BookRequestDto bookDto) {
         BookDto createdBook = bookService.save(bookDto);
         return ResponseEntity.ok(createdBook);
     }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<BookDto> updateBook(@PathVariable Long id, @RequestBody BookDto bookDto) {
+        bookDto.setId(id);
+        BookDto bookDto1 = bookService.updateBook(bookDto);
+        return ResponseEntity.ok(bookDto1);
```

Comment:
bookDto1 It is bad name for several reasons. On of them that 1 makes developer think that the may be other book dto`s and their order matters, but this in not the case